js/src/modals/AddContract/addContract.spec.js

```diff
@@ -20,36 +20,65 @@ import sinon from 'sinon';
 
 import AddContract from './';
 
-import { CONTRACTS, createApi } from './addContract.test.js';
+import { CONTRACTS, createApi, createRedux } from './addContract.test.js';
 
+let api;
 let component;
+let instance;
 let onClose;
+let reduxStore;
 
-function renderShallow (props) {
+function render (props = {}) {
+  api = createApi();
   onClose = sinon.stub();
+  reduxStore = createRedux();
+
   component = shallow(
     <AddContract
       { ...props }
       contracts={ CONTRACTS }
       onClose={ onClose } />,
-    {
-      context: {
-        api: createApi()
-      }
-    }
-  );
+    { context: { store: reduxStore } }
+  ).find('AddContract').shallow({ context: { api } });
+  instance = component.instance();
 
   return component;
 }
 
 describe('modals/AddContract', () => {
   describe('rendering', () => {
     beforeEach(() => {
-      renderShallow();
+      render();
     });
 
     it('renders the defauls', () => {
       expect(component).to.be.ok;
     });
   });
+
+  describe('onAdd', () => {
+    it('calls store addContract', () => {
+      sinon.stub(instance.store, 'addContract').resolves(true);
```

Comment:
I think in this case it's not worth it yet, but once we have more stubs, restoring them can easily be forgotten (which breaks the whole point of the stubs then). [sandboxes](http://sinonjs.org/docs/#sinon-sandbox) try to address this issue.